### PR TITLE
Update build

### PR DESCRIPTION
**Pull request by SuperQ**
Update build to the latest Prometheus build style.
* Use the Prometheus Makefile.common.
* Update to go modules.
* Update all vendoring.
* Use new promhttp package.
* Update promu config for Go modules.
* Convert to CircleCI

Signed-off-by: Ben Kochie <superq@gmail.com>

### CODE REVIEW

**Comment by SuperQ** [comment]
Marking WIP pending https://github.com/prometheus-community/stackdriver_exporter/pull/73

**Comment by simonpasquier** [comment]
LGTM though I don't see the job result from Circle CI. Did you enable this repo in Circle CI?

**Comment by SuperQ** [comment]
It's enabled, but it's not picking this up. It should trigger as soon as it hits master.

**Comment by SuperQ** [comment]
Oh, I figured it out, "build fork pull request" wasn't enabled in CircleCI.

File: .promu.yml [hunk 1]
```diff
@@ -1,29 +1,19 @@
 verbose: false
 go:
   cgo: false
+  # This must match .circle/config.yml.
+  version: 1.14
 repository:
   path: github.com/prometheus-community/stackdriver_exporter
 build:
-  flags: -a -tags 'netgo static_build'
+  flags: -mod=vendor -a -tags 'netgo static_build'
```
Comment (roidelapluie) [inline .promu.yml:9]: with go 1.14 this is optional

Comment (SuperQ) [reply .promu.yml:9]: Oh, thanks for the tip.

**Pull request merged**